lib/dartdoc.dart

```diff
@@ -273,19 +273,19 @@ class Dartdoc {
       referredFromElements.removeWhere((e) => !e.isCanonical);
     }
     if (warnOnElements != null) {
-      if (warnOnElements.any((e) => e.isCanonical)) {
-        warnOnElement = warnOnElements.firstWhere((e) => e.isCanonical);
-      } else {
-        // If we don't have a canonical element, just pick one.
-        warnOnElement = warnOnElements.isEmpty ? null : warnOnElements.first;
-      }
+      warnOnElement = warnOnElements.firstWhere((e) => e.isCanonical,
+          orElse: () => warnOnElements.isEmpty ? null : warnOnElements.first);
     }
 
     if (referredFromElements.isEmpty && referredFrom == 'index.html') {
       referredFromElements.add(packageGraph.defaultPackage);
     }
-    var message = warnOn;
-    if (referredFrom == 'index.json') message = '$warnOn (from index.json)';
+    String message;
```

Comment:
Why expand this?

Reply:
Ah sorry, it's sort of a habit style of mine, since we avoid potentially assigning `message` twice. I've reverted the change :)